src/Severity.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/*
+ * This file is part of Raven.
+ *
+ * (c) Sentry Team
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sentry;
+
+/**
+ * This class represents an enum of severity levels an event can be associated
+ * to.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+final class Severity
+{
+    /**
+     * This constant represents the "debug" severity level.
+     */
+    public const DEBUG = 'debug';
+
+    /**
+     * This constant represents the "info" severity level.
+     */
+    public const INFO = 'info';
+
+    /**
+     * This constant represents the "warning" severity level.
+     */
+    public const WARNING = 'warning';
+
+    /**
+     * This constant represents the "error" severity level.
+     */
+    public const ERROR = 'error';
+
+    /**
+     * This constant represents the "fatal" severity level.
+     */
+    public const FATAL = 'fatal';
+
+    /**
+     * This constant contains the list of allowed enum values.
+     */
+    public const ALLOWED_SEVERITIES = [
+        self::DEBUG,
+        self::INFO,
+        self::WARNING,
+        self::ERROR,
+        self::FATAL,
+    ];
+
+    /**
+     * @var string The value of this enum instance
+     */
+    private $value;
+
+    /**
+     * Constructor.
+     *
+     * @param string $value The value this instance represents
+     */
+    private function __construct(string $value = self::INFO)
+    {
+        if (!\in_array($value, self::ALLOWED_SEVERITIES, true)) {
+            throw new \InvalidArgumentException(sprintf('The "%s" is not a valid enum value.', $value));
+        }
+
+        $this->value = $value;
+    }
+
+    /**
+     * Creates a new instance of this class with the given severity value.
+     *
+     * @param string $severity The severity
+     *
+     * @return self
+     */
+    public static function fromString(string $severity): self
```

Comment:
Well but it's the same of the constructor, it doesn't give you any addition nor translation to it..

Reply:
I think that `Severity::fromString(...)` is more readable than having to do in-place (because most of the cases you will need to create this object imho is inline code) `new Severity(...)`. I will remove this method anyway